packages/extension-store-locator/src/setup-app.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React from "react";
+import loadable from "@loadable/component";
+import {
+  ApplicationExtension,
+  IRouteConfig,
+} from "@salesforce/pwa-kit-react-sdk/ssr/universal/extensibility";
+import {withOptionalChakra} from "*/components/store-locator/with-optional-chakra-provider";
+import { withStoreLocator } from "*/components/store-locator/with-store-locator";
+import { ReactExtensionConfig as Config } from "./types";
+
+const StoreLocator = loadable(() => import("*/pages/store-locator"));
+
+class StoreLocatorExtension extends ApplicationExtension<Config> {
+  DEFAULT_PATH = "/store-locator";
+  REQUIRED_CONFIG_FIELDS = ['defaultCountry', 'defaultCountryCode', 'defaultDistance', 'defaultDistanceUnit', 'defaultPageSize', 'defaultPostalCode', 'supportedCountries']
+
+  extendApp(App: React.ComponentType): React.ComponentType {
+    const config = this.getConfig();
+
+    const missingFields = this.REQUIRED_CONFIG_FIELDS.filter(field => !config[field]);
+    if (missingFields.length) {
+      throw new Error(`Missing required config fields: ${missingFields.join(', ')}`);
+    }
```

Comment:
This runtime argument validation should be generalized across all application extension development. We should look into integrating an external library like `zod` so that we can use our extensions config to create a schema and use it to validate at runtime. 

Can you create a ticket to do this, potentially file that under the "future" work. AC should be something about adding this to our base extension boilerplate.

Reply:
@kevinxh to create a ticket ☝️